model/lifecycle.go

```diff
@@ -969,3 +969,105 @@ func sortLayer(layer []task.Task, idToDisplayName map[string]string) []task.Task
 	}
 	return sortedLayer
 }
+
+// Given a patch version and a list of variant/task pairs, creates the set of new builds that
+// do not exist yet out of the set of pairs. No tasks are added for builds which already exist
+// (see AddNewTasksForPatch).
+func AddNewBuilds(activated bool, v *version.Version, p *Project, tasks TaskVariantPairs) error {
```

Comment:
I'd put the arguments in the following order: (project, version, tasks, activated)